### PR TITLE
RunNoShellQA fixes (HPC-7645)

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

**Comment by wdpypere** [comment]
version bump, jira link?

**Comment by boegel** [comment]
@stdweird No enhanced test to cover the fix?

**Comment by stdweird** [comment]
@boegel tests modified to cover more noshell usage

File: test/run.py [hunk 1]
```diff
@@ -38,7 +38,14 @@
 import shutil
 from unittest import TestLoader, main
 
-from vsc.utils.run import CmdList, run, run_simple, run_asyncloop, run_timeout, RunQA, RunTimeout
+#import logging
+#logging.basicConfig(level=logging.DEBUG)
```
Comment (boegel) [inline test/run.py]: @stdweird Remove these, since they're not used?

File: test/run.py [hunk 2]
```diff
@@ -227,6 +243,14 @@ def test_qa_regex(self):
         ec, output = run_qas([sys.executable, SCRIPT_QA, 'whattime'], qa_reg=qa_dict)
         self.assertEqual(ec, 0)
 
+    def test_qa_legacy_regex(self):
+        """Test regex based q and a (works only for qa_reg)"""
+        qa_dict = {
+                   '\s(?P<time>\d+(?:\.\d+)?).*?What time is it\?': '%(time)s',
+                   }
```
Comment (boegel) [inline test/run.py]: I don't see the point in spreading this across multiple lines?

And if you insist, just indent 2nd line with 4 spaces compared to `qa_dict = {`?

**Comment by boegel** [comment]
@stdweird Can you make the PR title/descr a bit more descriptive? It's not really clear to me what's being fixed here, to be honest...

**Comment by stdweird** [comment]
@boegel title fixed

**Comment by boegel** [comment]
@stdweird What about the other minor remarks?

File: test/run.py (hunk 1 above)
Comment (stdweird) [reply test/run.py]: i can't enable them by default, it screws up some of the tests; but i'll uncomment the import logging :wink:

File: test/run.py (hunk 2 above)
Comment (stdweird) [reply test/run.py]: it tests something else, i'd rathe rnot mix it up
i'll reindent

**Pull request merged**